packages/xstate-fsm/src/types.ts

```diff
@@ -104,9 +108,7 @@ export namespace StateMachine {
     changed?: boolean | undefined;
     matches: <TSV extends TState['value']>(
       value: TSV
-    ) => this is TState extends { value: TSV }
-      ? TState & { value: TSV }
-      : never;
+    ) => this is { value: TSV; context: TContext };
```

Comment:
what motivated this change? could you give an example of what didn't work before this change?

Reply:
Since TSV was extended with TSstate['value'] this worked well for a true expression. But with a false expression, the context became type "never". So it was not possible to check multiple states: 
if(state.matches('idle')) {}
else if(state.matches('latest')) {}

Reply:
could you add a test case for this [here](https://github.com/statelyai/xstate/blob/3d3a385a655ef524f2c90d908c04ed9855c42b71/packages/xstate-fsm/test/types.test.ts#L3)?

Reply:
There is no need to add new examples there. This test passes in the current implementation because the value is of type any. You can pass the number 12 to the match function and it will work. But I narrowed the value down to a string, so the typescript started swearing at the second branch of the condition. Then I changed the types for the correct behavior.